src/main/java/com/puppycrawl/tools/checkstyle/api/AutomaticBean.java

```diff
@@ -53,7 +53,8 @@
  * calling the bean's setters for all configuration attributes.
  * @author lkuehne
  */
-public class AutomaticBean
+// -@cs[AbstractClassName] class name cannot be changed
```

Comment:
please define why it can not be changed, "because it is API" ?
should we make AbstractAutomaicBean ? and deprecate AutomaticBean ?

Reply:
> "because it is API" ?

Yes.

> should we make AbstractAutomaicBean ? and deprecate AutomaticBean ?

I am fine with this change.
Since we are changing name, do we still want to keep `AutomaticBean`?

Reply:
hmmm, please propose a name ... and lets move this to separate issue as it big update it should explicit and ease to catch in our releasenotes.